### DISCUSSION TITLE
Unable to get YouTube URLs to interact with models - API and setup all ok

### DISCUSSION THREAD

**Discussion by fearby** [danielmiessler/Fabric#577] [Q&A] [answered]
Yes a hobbyist with little experience.
Installed on MacOS ok and API keys added except for Google API
The example on the Github landing page will just give me nothing and the following error:

yt --transcript https://youtube.com/watch?v=uXs-zPc63kM | fabric --stream --pattern extract_wisdom
zsh: no matches found: https://youtube.com/watch?v=uXs-zPc63kM
I'm sorry, but it seems like there was no input provided for me to process. Could you please provide the text content you'd like me to analyze?%

What am I doing wrong?

**Comment by kayozxo** [comment]
Try `yt --transcript youtube_video_link | fabric -sp extract_wisdom`

**Reply by fearby** [reply]
Just get the same response: "I'm sorry, but it seems like there's no input provided for me to process. Could you please provide the text content you'd like me to analyze?"

Tried reinstalling Fabric all and putting new API keys in but always the same response

**Reply by kayozxo** [reply]
Check your `~/.zshrc` or `~/.bashrc` files. You should find these lines at the bottom of these files. If not, then add them.
![image](https://github.com/danielmiessler/fabric/assets/114288044/cc815b8b-7491-4b41-b8d7-cac27c8fadff)
If the issue persists, try contacting the devs.

**Reply by fearby** [reply]
My ~/.zshrc is fine its just that the yt function just doesn't work for some reason

**Reply by kayozxo** [reply]
try `(yt --transcript youtube_link) | fabric -sp extract_wisdom` or `(yt --transcript <youtube_link>) | fabric -sp extract_wisdom`

**Reply by tomcoady** [reply]
After successfully adding quotes to the YT url is suspected fabric, so uninstalled, reinstalled but still getting 'NoneType":

`Successfully` installed bcrypt-4.2.0 decorator-5.1.1 deprecated-1.2.14 fabric-3.2.2 invoke-2.2.0 paramiko-3.4.1 pynacl-1.5.0 wrapt-1.16.0
echo "Hello, world" | fabric --pattern extract_wisdom
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1723884929.462370 3973512 check_gcp_environment_no_op.cc:29] ALTS: Platforms other than Linux and Windows are not supported
Error: 'NoneType' object has no attribute 'chat'
'NoneType' object has no attribute 'chat'

**Comment by RoeslerM** [comment]
I face same problem. 
yt is unable to fetch the URL of the video. I tried multiples way. 
I had played around a bit and found that the Youtube ID of the video sometimes worked, but I could not get a constant reproducible result for this, so I meanwhile use the transcript function from Youtube, cut&paste that and pipe it to fabric.
Not very convenient, but works

**Comment by RoeslerM** [comment]
Ah, a colleague of mine with the same issue found a workaround.
Put the URL in "", then YT works

**Reply by fearby** [reply]
Tried that already.

When I just use yt in the fabric directory I get:
\usage: yt [-h] [--config CONFIG] [--paste] [--paste-detailed] [--detailed]
          [--rpdb] [--parallel]
          {help,instinfo,version,load,mapserver,pastebin,pastebin_grab,plot,rpdb,stats,update,delete_image,upload_image,upload,config,search,download}
          ...
Does that mean it is a different version of yt to that needed by fabric?

**Reply by RoeslerM** [reply]
can you please check "type yt" ?
it seems to me that you are not calling the right program. Mayb you have another file from an other application, that is called first in your path

when call yt, I get:

usage: yt [-h] [--duration] [--transcript] [--comments] [--lang LANG] url
yt: error: the following arguments are required: url

**Reply by fearby** [reply]
I get:

usage: yt [-h] [--config CONFIG] [--paste] [--paste-detailed] [--detailed]
          [--rpdb] [--parallel]
          {help,instinfo,version,load,mapserver,pastebin,pastebin_grab,plot,rpdb,stats,update,delete_image,upload_image,upload,config,search,download}
          ...

yt command line arguments

options:
  -h, --help            show this help message and exit
  --config CONFIG       Set configuration option, in the form param=value
  --paste               Paste traceback to paste.yt-project.org
  --paste-detailed      Paste a detailed traceback with local variables to
                        paste.yt-project.org
  --detailed            Display detailed traceback.
  --rpdb                Enable remote pdb interaction (for parallel
                        debugging).
  --parallel            Run in MPI-parallel mode (must be launched as an MPI
                        task)

subcommands:
  Valid subcommands

  {help,instinfo,version,load,mapserver,pastebin,pastebin_grab,plot,rpdb,stats,update,delete_image,upload_image,upload,config,search,download}
    help                Print help message
    instinfo            Get some information about the yt installation
    version             Get some information about the yt installation
    load                Load a single dataset into an IPython instance
    mapserver           Serve a plot in a GMaps-style interface
    pastebin            Post a script to an anonymous pastebin
    pastebin_grab       Print an online pastebin to STDOUT for local use.
    plot                Create a set of images
    rpdb                Connect to a currently running (on localhost) rpd
                        session. Commands run with --rpdb will trigger an rpdb
                        session with any uncaught exceptions.
    stats               Print stats and max/min value of a given field (if
                        requested), for one or more datasets (default field is
                        density)
    update              Update the yt installation to the most recent version
    delete_image        Delete image from imgur.com.
    upload_image        Upload an image to imgur.com. Must be PNG.
    upload              Upload a file to yt's curldrop.
    config              Get and set configuration values for yt
    search              Attempt to find outputs that yt can recognize in
                        directories.
    download            Download a file from http://yt-project.org/data and
                        save it to a particular location. Files can be saved
                        to the locations provided by the "test_data_dir" or
                        "supp_data_dir" configuration entries, or any valid
                        path to a location on disk.


And when I ask about instinfo I get:

yt module located at:
    /Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages

**Comment by RoeslerM** [comment]
So you obviously call a program that is called „yt“ which does not belong to the fabric package.you can find out where the wrong „yt“ is located by entering “type yt“You can also search your entire hard disk by entering „find / -name yt -print“This should result in more then one hitsyou then can rename the wrong yt or alternatively call the correct yt with absolute pathTypos (C) by iPhoneAm 11.06.2024 um 18:29 schrieb Simon Fearby ***@***.***>:﻿
I get:
usage: yt [-h] [--config CONFIG] [--paste] [--paste-detailed] [--detailed]
[--rpdb] [--parallel]
{help,instinfo,version,load,mapserver,pastebin,pastebin_grab,plot,rpdb,stats,update,delete_image,upload_image,upload,config,search,download}
...
yt command line arguments
options:
-h, --help            show this help message and exit
--config CONFIG       Set configuration option, in the form param=value
--paste               Paste traceback to paste.yt-project.org
--paste-detailed      Paste a detailed traceback with local variables to
paste.yt-project.org
--detailed            Display detailed traceback.
--rpdb                Enable remote pdb interaction (for parallel
debugging).
--parallel            Run in MPI-parallel mode (must be launched as an MPI
task)
subcommands:
Valid subcommands
{help,instinfo,version,load,mapserver,pastebin,pastebin_grab,plot,rpdb,stats,update,delete_image,upload_image,upload,config,search,download}
help                Print help message
instinfo            Get some information about the yt installation
version             Get some information about the yt installation
load                Load a single dataset into an IPython instance
mapserver           Serve a plot in a GMaps-style interface
pastebin            Post a script to an anonymous pastebin
pastebin_grab       Print an online pastebin to STDOUT for local use.
plot                Create a set of images
rpdb                Connect to a currently running (on localhost) rpd
session. Commands run with --rpdb will trigger an rpdb
session with any uncaught exceptions.
stats               Print stats and max/min value of a given field (if
requested), for one or more datasets (default field is
density)
update              Update the yt installation to the most recent version
delete_image        Delete image from imgur.com.
upload_image        Upload an image to imgur.com. Must be PNG.
upload              Upload a file to yt's curldrop.
config              Get and set configuration values for yt
search              Attempt to find outputs that yt can recognize in
directories.
download            Download a file from http://yt-project.org/data and
save it to a particular location. Files can be saved
to the locations provided by the "test_data_dir" or
"supp_data_dir" configuration entries, or any valid
path to a location on disk.

**Comment by RoeslerM** [comment]
looks like this is your „yt“Ubuntu Manpage: yt - command-line tool allowing to access some of yt's basic functionality without openingmanpages.ubuntu.comTypos (C) by iPhoneAm 11.06.2024 um 19:23 schrieb Martin Rösler ***@***.***>:﻿So you obviously call a program that is called „yt“ which does not belong to the fabric package.you can find out where the wrong „yt“ is located by entering “type yt“You can also search your entire hard disk by entering „find / -name yt -print“This should result in more then one hitsyou then can rename the wrong yt or alternatively call the correct yt with absolute pathTypos (C) by iPhoneAm 11.06.2024 um 18:29 schrieb Simon Fearby ***@***.***>:﻿
I get:
usage: yt [-h] [--config CONFIG] [--paste] [--paste-detailed] [--detailed]
[--rpdb] [--parallel]
{help,instinfo,version,load,mapserver,pastebin,pastebin_grab,plot,rpdb,stats,update,delete_image,upload_image,upload,config,search,download}
...
yt command line arguments
options:
-h, --help            show this help message and exit
--config CONFIG       Set configuration option, in the form param=value
--paste               Paste traceback to paste.yt-project.org
--paste-detailed      Paste a detailed traceback with local variables to
paste.yt-project.org
--detailed            Display detailed traceback.
--rpdb                Enable remote pdb interaction (for parallel
debugging).
--parallel            Run in MPI-parallel mode (must be launched as an MPI
task)
subcommands:
Valid subcommands
{help,instinfo,version,load,mapserver,pastebin,pastebin_grab,plot,rpdb,stats,update,delete_image,upload_image,upload,config,search,download}
help                Print help message
instinfo            Get some information about the yt installation
version             Get some information about the yt installation
load                Load a single dataset into an IPython instance
mapserver           Serve a plot in a GMaps-style interface
pastebin            Post a script to an anonymous pastebin
pastebin_grab       Print an online pastebin to STDOUT for local use.
plot                Create a set of images
rpdb                Connect to a currently running (on localhost) rpd
session. Commands run with --rpdb will trigger an rpdb
session with any uncaught exceptions.
stats               Print stats and max/min value of a given field (if
requested), for one or more datasets (default field is
density)
update              Update the yt installation to the most recent version
delete_image        Delete image from imgur.com.
upload_image        Upload an image to imgur.com. Must be PNG.
upload              Upload a file to yt's curldrop.
config              Get and set configuration values for yt
search              Attempt to find outputs that yt can recognize in
directories.
download            Download a file from http://yt-project.org/data and
save it to a particular location. Files can be saved
to the locations provided by the "test_data_dir" or
"supp_data_dir" configuration entries, or any valid
path to a location on disk.

**Reply by fearby** [reply]
Thanks - I did have a thing called MacYTDL which it must have been calling but I have uninstalled that and reinstalled Fabric as per instructions and... still get the same initial error which I think is an API problem as it seems to be openai is not able to handle YouTube links:

yt --transcript https://youtube.com/watch?v=uXs-zPc63kM | fabric --stream --pattern extract_wisdom
zsh: no matches found: https://youtube.com/watch?v=uXs-zPc63kM
I'm sorry, but it seems like there was no input provided for me to process. Could you please provide the text content you'd like me to analyze?%

**Reply by RoeslerM** [reply]
ok, you say fabric is working, so lets drop the fabric part.
Only focus on the yt part.
when you enter "type yt", what do you get as result ?

**Reply by fearby** [reply]
Thanks - 

yt is /Library/Frameworks/Python.framework/Versions/3.12/bin/yt

**Comment by RoeslerM** [comment]
ok, so you still have a mess between your python installation and fabric. Maybe your deinstallation did not go through properly or you have no virtual python environments installed.

Never the less: give this a try: rename the yt program to yt_backup

**sudo mv /Library/Frameworks/Python.framework/Versions/3.12/bin/yt /Library/Frameworks/Python.framework/Versions/3.12/bin/yt_backup**

and now try yt from fabric again:

**yt --transcript https://youtube.com/watch?v=uXs-zPc63kM** 

you should at least get a different error message now. 
In worst case: you get something like 
**zsh: command not found: yt**
In this case yt did not got installed correctly and you have a bigger problem

If you get an error 
**zsh: no matches found: https://youtube.com/watch?v=uXs-zPc63kM**
then yt at least seems to work, but the youtube file was not found.

**Reply by fearby** [reply]
Ok still same error -zsh: no matches found: https://youtube.com/watch?v=uXs-zPc63kM

**Comment by RoeslerM** [comment]
ok, now try to mask the URL with "".
try 
yt --transcript "https://youtube.com/watch?v=uXs-zPc63kM"

**Reply by timothyjoh** [reply, marked as answer]
I also had this problem, wrapping the URL in quotes works

**Reply by fearby** [reply]
Had a hardware problem so now on a fresh install and new logic board and the URL quotes now works.

**Reply by fearby** [reply]
As an addendum I am back where I started again even with URL in quotes.

I think it is a conflict with facefusion which I also have subsequently installed and now get the same errors as I started with and can't get YouTube to interact.
